### ISSUE TITLE
[Feature request]: able to take snapshot of all movies and restore to the taken state

### ISSUE THREAD

**Issue by the-liquid-metal** [vzhd1701/gridplayer#19]
### Suggestion

There is annoying task to set all movie time position and zoom level (and may be other things) to a particular state again and again while we analize them. It would be better if we can take a snapshot and than restore it whenever we need.

**Comment by vzhd1701** [comment]
Do you mean like saving multiple states for single video and then quickly restoring them with Ctrl+1,2,3.. for example?

**Comment by the-liquid-metal** [comment]
No, but single state of all videos in a single session. Like save the playlist, close it, and then reopen it again. A single combination key would be enough, eg. CTRL+1. But if we can save multiple snapshot will make it great.

To restore the session, logically by CTRL+SHIFT+1, or friends.

**Comment by the-liquid-metal** [comment]
This feature is related to #24 (Bring VLC adjustments and effects into GridPlayer enhancement )

**Comment by the-liquid-metal** [comment]
As a note, if we hit CTRL+1, then play the movies, whether it only one or all, then pause gain, then hit CTRL+1 again, the previous snapshot is overiden by the latest one.

Snapshot functionality is like CTLR+C functionallity in notepad or any text editor: only the last hit will be available in clipboard.

If multiple snapshots is achieved, the overiden snapshot only affected to the corresponding key:
If i hit CTRL+3, then only snapshot containend in this key is overiden, i.e CTRL+1, CTRL+2, CTRL+4, and so on is not affected.

**Comment by the-liquid-metal** [comment]
After viewing the contents of the *.gpls file, it is quite simple to save the snapshots without intervening the already achieved functionality. There is already "current_position" key as a representative of where the movie has to start. The "current_position" value is just a plain integer.

logically, snapshot contained in CTRL+1 will be named as "snapshot_1", snapshot contained in CTRL+2 will be named as  "snapshot_2", and so on.

**Comment by vzhd1701** [comment]
Snapshots support added! Please check out the new version.

![image](https://user-images.githubusercontent.com/5212766/175575894-ea4819c5-2f6c-4aef-a6a7-7c97dd01282e.png)
